### PR TITLE
fix: NativeAOT inApp setting

### LINKED ISSUE #2817: Native AOT in-app frames
Check what we sent as in-app from the runtime and how we show it on the server.

The thing is, we don't really know at runtime whether the frame is in-app or not.

If there's any change, we should try to do it prior to 4.0.0 release because it would be breaking in grouping.

### PR DESCRIPTION
**Pull request by vaind**
closes #2817 

#skip-changelog the original code hasn't been released yet (outside of alpha)

### CODE REVIEW

File: src/Sentry/SentryStackFrame.cs [hunk 1]
```diff
@@ -182,7 +182,7 @@ public void ConfigureAppFrame(SentryOptions options)
         {
             ConfigureAppFrame(options, Function, mustIncludeSeparator: true);
         }
-        else
+        else if (ImageAddress is null or 0 && InstructionAddress is null or 0) // Leave InApp=null on NativeAOT
```
Comment (jamescrosswell) [inline src/Sentry/SentryStackFrame.cs:185]: Just curious... why does this condition imply it's not a NativeAOT app?

Comment (vaind) [reply src/Sentry/SentryStackFrame.cs:185]: https://github.com/getsentry/sentry-dotnet/blob/feat/4.0.0/src/Sentry/Internal/DebugStackTrace.cs#L237

Comment (bitsandfoxes) [reply src/Sentry/SentryStackFrame.cs:185]: I'm not all the way up to speed but a simple `0` works? The framework returns an `IntPtr`, the `InstructionAddress` is a `nint`.

**Review by bitsandfoxes — APPROVED** [review]

File: src/Sentry/SentryStackFrame.cs (hunk 1 above)
Comment (vaind) [reply src/Sentry/SentryStackFrame.cs:185]: it's nullable so I have to check for that first

**Pull request merged**